libraries/azure_backend.rb

```diff
@@ -0,0 +1,794 @@
+  def resource_short(opts)
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+    url = "#{@azure.resource_manager_endpoint_url}subscriptions/#{@azure.credentials[:subscription_id]}/resources"
+    params = {
+      '$filter' => Helpers.odata_query(opts),
+      '$expand' => Helpers.odata_query(%w{createdTime changedTime provisioningState}),
+    }
+    api_version = @azure.resource_manager_endpoint_api_version
+    short_description, suggested_api_version = rescue_wrong_api_call(url, api_version, params)
+    # If suggested_api_version is not nil, then the resource manager api version should be updated.
+    unless suggested_api_version.nil?
+      @resource_manager_endpoint_api = suggested_api_version
+      Inspec::Log.warn "Resource manager endpoint api version should be updated with #{suggested_api_version} in"\
+        ' `libraries/backend/helpers.rb`'
+    end
+    short_description[:value] || []
+  end
+
+  # @return [Array] [ HTTP_response_body, api_version_suggested ]
+  # @param url [String] The url without any parameters or headers.
+  # @param api_version [String] The api version of the target Azure API endpoint, `2020-06-01`.
+  # @param params [Hash] The query parameters without the api version.
+  #
+  # @note
+  #   - HTTP_response_body will be in JSON with symbolized keys.
+  #
+  # @example Result
+  #   [{:value => 'blah'}, '2020-01-01']
+  def rescue_wrong_api_call(url, api_version, params = {})
+    begin
+      response = @azure.rest_get_call(url, api_version, params)
+    rescue UnsuccessfulAPIQuery::UnexpectedHTTPResponse::InvalidApiVersionParameter => e
+      api_version_suggested = e.suggested_api_version(api_version)
+      Inspec::Log.warn "Incompatible api version: #{api_version}\n"\
+        "Trying with the latest api version suggested by the Azure Rest API: #{api_version_suggested}."
+      if api_version_suggested.nil?
+        Inspec::Log.warn 'Failed to acquire suggested api version from the Azure Rest API.'
+      else
+        response = @azure.rest_get_call(url, api_version_suggested, params)
+      end
+    end
+    [response, api_version_suggested]
+  end
+
+  # Construct resource_id from the provided parameters.
+  #
+  # @return [String] The resource_id of an Azure cloud resource.
+  # "/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/
+  #   Microsoft.Compute/virtualMachines/{resource_name}"
+  #
+  # @note: resource_provider should include parent resource path if there is, e.g.:
+  #   Microsoft.Compute/virtualMachines/extensions
+  # @see https://docs.microsoft.com/en-us/rest/api/resources/resources/get
+  #
+  def construct_resource_id
+    required_arguments = %i(resource_group name resource_provider)
+    raise ArgumentError, "Following parameters should be provided to construct a resource_id: #{required_arguments}" \
+      unless required_arguments.all? { |r_p| @opts.keys.include?(r_p) }
```

Comment:
would be nice to change `r_p` I assumed this is short for `resource_provider`. I think simply `ra_key` or `key` would make sense